projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * WordAds Consent Management Provider
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Class WordAds_Consent_Management_Provider
+ *
+ * This is an IAB TCF v2.2 compatible implementation of a [Consent Management Provider](https://iabeurope.eu/transparency-consent-framework/)
+ * to comply with GDPR requirements for privacy and transparency related to advertising.
+ *
+ * The repository for the CMP front-end JS can be found at: https://github.com/Automattic/cmp
+ * Build with `yarn build` then create a Phabricator patch to deploy the assets in the `/dist` folder to WPCOM.
+ */
+class WordAds_Consent_Management_Provider {
+
+	/**
+	 * IAB specified cookie name for storing the consent string.
+	 */
+	const COOKIE_NAME = 'euconsent-v2';
+
+	/**
+	 * Initializes loading of the frontend framework.
+	 */
+	public static function init() {
+		// Prevent Cookies & Consent banner from displaying when the CMP is active.
+		add_filter( 'jetpack_disable_eu_cookie_law_widget', '__return_true' );
```

Comment:
Should we support our Cookie Consent block too?

Reply:
Would you want to block the display of the block on the frontend only, or in the editor as well?

Reply:
The frontend at minimum, but preferably in the editor as well. That would be the least confusing option for users, I think.

Reply:
To stop things on the frontend, you could add an early return here if a specific condition is met:
https://github.com/Automattic/jetpack/blob/8447ef5075727869218037724844a6b6ba29b35c/projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php#L71

In the editor, I'm guessing you could display a notice somewhere here:
https://github.com/Automattic/jetpack/blob/79a546280f22e957c9792832f414645d50a3eba1/projects/plugins/jetpack/extensions/blocks/cookie-consent/edit.js#L82